File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.py
class Solution:
    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
        total = 0
        sums = dict()

        for i in range(len(nums)):
            total += nums[i]
            total %= k

            if total == 0 and i > 0: return True
            if total in sums and i - sums[total] > 1: return True
            if total not in sums: sums[total] = i

        return False

File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.java
class Solution {
    public boolean checkSubarraySum(int[] nums, int k) {
        Map<Integer, Integer> map = new HashMap<>();
        int total = 0;

        for (int i = 0; i < nums.length; i++) {
            total += nums[i];
            total %= k;

            if (total == 0 && i > 0) return true;
            if (map.containsKey(total) && i - map.get(total) > 1) return true;
            if (!map.containsKey(total)) map.put(total, i);
        }

        return false;
    }
}